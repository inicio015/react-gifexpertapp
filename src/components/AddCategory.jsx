import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {


  const [inputValue, setInputValue] = useState()
   

  /*Desestructuramos el target del evento que viene  del onchange*/ 
  const onInputChanged = ({target}) => {
    setInputValue(value => target.value);
  }
  const onSubmit = (event) => {
    event.preventDefault();

    //ESTA FUNCION SE CUMPLE SI TENGO MAS DE UN CARACTER
    if(inputValue.trim().length <= 1) return; 
    
    //setCategories( categories => [inputValue,...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
    
  }


  return (
    /*Mientras el elemento tenga un contenedor padre no tendremos problemas
    en agregarlo en un mismo componente, pero cuando el elemento tengo un hermano
    entonces en ese momento es mejor crear otr fragmento e importarlo */


    /*Coloamos un valor por defecto en value */ 
    /*Colocamos el evento onchange y asignamos un nuevo evento*/ 
    /*Al igual que en vue o cualquier cosa con javascript con solo colocar el nombre de la funcion
    obtenemos su evento directamente */ 

    
    <form onSubmit={ onSubmit }>
        <input 
            type="text" 
            placeholder="Buscar gifs"
            value= {inputValue || ''}
            onChange={ onInputChanged }
        />
    </form>
  
  )
}
