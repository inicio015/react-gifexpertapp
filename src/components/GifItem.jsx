import PropTypes from 'prop-types';
export const GifItem = ({title, url,id}) => {
  return (
    <div className="card-style col-3">
        <img src={url} alt={ title } />
        <p>{ title }</p>
        
    </div>
  )
}


//Tipos de propiedades de  nuestro componente y colocar el tipo y si es requerido o lo que se espera de ese proptype 
// siempre debemos a√±adir los proptypes a las dependencias en este caso descargarlo
// yarn add prop-types
GifItem.propTypes = {
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired,
  //id:PropTypes.number.isRequired
}
