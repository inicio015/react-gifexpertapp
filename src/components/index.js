export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';

export { useState } from 'react';
