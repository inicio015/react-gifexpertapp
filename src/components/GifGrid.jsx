import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  
  //HOOKS EN REACT BRUTALL ASI REUTILIZAMOS CODIGO, NUESTRA LOGICA EN OTRA PARTE COMO TAL 
  const {images, isLoading} = useFetchGifs(category)

  //No se usa la palabra reservada class si no que className
  return (
    <>
      <div className='row'>
        <hr />
        <h3>{category}</h3>

        {
          //if corto es fino y util
          isLoading && ( <h2>Cargando...</h2>)
        }
        <div className="card-grid col">
        {   
            images.map( (image) => (
                
                <GifItem 
                  key={image.id} 

                  //puedo repartir todas las propiedades de este modo y asi le vamos a decir que reciba todas las propertis de manera mas facil
                  //brutal 
                  { ...image }
                />
                
              )
            ) 
          }

        </div> 
      </div>
    </> 
  )
}
