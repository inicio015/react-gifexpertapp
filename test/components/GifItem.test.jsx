import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe(' Pruebas en el componente gifitem',() => {
    const title = "SAITAMA";
    const url = "https://one-punch.com/saitama.jpg";
    test('Probando este componente de GifItem y debe hacer match con el snapshot',() => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();


    })
})