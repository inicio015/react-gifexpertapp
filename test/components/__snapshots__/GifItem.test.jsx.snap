// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en el componente gifitem Probando este componente de GifItem y debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card-style col-3"
  >
    <img
      alt="SAITAMA"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      SAITAMA
    </p>
  </div>
</div>
`;
